capellambse_context_diagrams/collectors/class_tree.py

```diff
@@ -0,0 +1,88 @@
+# SPDX-FileCopyrightText: 2022 Copyright DB Netz AG and the capellambse-context-diagrams contributors
+# SPDX-License-Identifier: Apache-2.0
+"""This submodule defines the collector for the Class-Tree diagram."""
+from __future__ import annotations
+
+import collections.abc as cabc
+import typing as t
+
+from capellambse import helpers
+from capellambse.model.crosslayer import information
+
+from .. import _elkjs, context
+from . import generic, makers
+
+LAYOUT_OPTIONS: _elkjs.LayoutOptions = {
+    "algorithm": "layered",
+    "edgeRouting": "ORTHOGONAL",
+    "elk.direction": "DOWN",
+    "partitioning.activate": True,
+    "edgeLabels.sideSelection": "ALWAYS_DOWN",
+}
+
+
+def collector(
+    diagram: context.ContextDiagram, params: dict[str, t.Any] | None = None
+) -> _elkjs.ELKInputData:
+    """Return the class tree data for ELK."""
+    params = params or {}
+    assert isinstance(diagram.target, information.Class)
+    data = generic.collector(diagram, no_symbol=True)
+    if params.get("partitioning", False):
+        data["children"][0]["layoutOptions"] = {}
+        data["children"][0]["layoutOptions"]["elk.partitioning.partition"] = 0
+    data["layoutOptions"] = LAYOUT_OPTIONS
+    data["layoutOptions"]["algorithm"] = (params or {})["algorithm"]
+    data["layoutOptions"]["elk.direction"] = (params or {})["direction"]
+    data["layoutOptions"]["edgeRouting"] = (params or {})["edgeRouting"]
```

Comment:
This global `LAYOUT_OPTIONS` default dictionary doesn't seem to be necessary at all, if I'm getting this right: ClassTreeDiagram always builds a full `params` dict with every possible key. This ultimately gets through to here, where it gets translated to ELK's config keys. That means, `params` will always be passed explicitly with a (non-empty) value.

So, the `| None = None` fallback/default in the function signature is not needed, as is the `params = params or {}` and these `(params or {})` parts here.